lib/task_manager.js

```diff
@@ -0,0 +1,49 @@
+class TaskManager {
+  constructor(startBlock, blockInterval) {
+    this.map = {};
+    this.queue;
+    this.buffer = [];
+    this.blockInterval = blockInterval;
+    this.currentInterval = {
+      fromBlock: startBlock + 1,
+      toBlock: startBlock + blockInterval
+    };
+  }
+
+  async initPQueue(maxConcurrentRequests) {
```

Comment:
We should not expose internal implementation to client. PQueue is too specific. Could you change method name?